ArrCharOps.java

```diff
@@ -127,6 +170,31 @@ public static long hashCode(char[] arr) {
      */
     public static int compareTo(String str1, String str2) {
         // Replace the following statement with your code
+        if (str1 == null || str2 == null)
+            return -2;
+        if (str1.isEmpty() && str2.isEmpty())
+            return 0;
+        else if (str1.isEmpty())
+            return -2;
+        else if (str2.isEmpty())
+            return -2;
```

Comment:
could have joined them with || 
if you want to use tools from course `str1.length == 0` would be good